From 9c5ced4092f6a885809a85144cdd936d172041d1 Mon Sep 17 00:00:00 2001
From: Nana GUo <nana.n.guo@intel.com>
Date: Thu, 2 Jan 2014 08:10:41 -0500
Subject: [PATCH 09/17] Add libjpeg-turbo integration to skia

BZ: 162001

Add switch between libjpeg-turbo and libjpeg

Change-Id: Ibdaa244976919efff58c140cdbb1ce1131e4d013
Category: aosp improvement
Domain: Video.Media-jpeg
Origin: internal
Upstream-Candidate: no, not-aosp
Signed-off-by: Nana GUo <nana.n.guo@intel.com>
---
 Android.mk | 18 ++++++++++++++++--
 1 file changed, 16 insertions(+), 2 deletions(-)

diff --git a/Android.mk b/Android.mk
index 757b76f..a4a64ba 100644
--- a/Android.mk
+++ b/Android.mk
@@ -584,7 +584,6 @@ LOCAL_SHARED_LIBRARIES := \
 	libdl \
 	libcutils \
 	libft2 \
-	libjpeg \
 	libpng \
 	libutils \
 	libz \
@@ -592,6 +591,14 @@ LOCAL_SHARED_LIBRARIES := \
 	libutils \
 	libEGL \
 	libGLESv2
+ifeq ($(USE_LIBJPEG_TURBO),true)
+LOCAL_SHARED_LIBRARIES += \
+	libjpeg-turbo
+else
+LOCAL_SHARED_LIBRARIES += \
+	libjpeg
+endif
+
 
 LOCAL_STATIC_LIBRARIES := \
 	libgif \
@@ -630,11 +637,18 @@ LOCAL_C_INCLUDES := \
 	external/zlib \
 	external/libpng \
 	external/giflib \
-	external/jpeg \
 	external/webp/include \
 	frameworks/base/opengl/include \
 	external/expat/lib
 
+ifeq ($(USE_LIBJPEG_TURBO),true)
+LOCAL_C_INCLUDES += \
+	$(TARGET_OUT_HEADERS)
+else
+LOCAL_C_INCLUDES += \
+	external/jpeg
+endif
+
 LOCAL_EXPORT_C_INCLUDE_DIRS := \
 	$(LOCAL_PATH)/include/config \
 	$(LOCAL_PATH)/include/core \
-- 
2.7.4

